### PR TITLE
adjust pio layout if pe layout is changed

### LINKED ISSUE #1192: PIO settings not adjusted with pe changes
Currently both PIO_STRIDE and PIO_NUMTASKS is set at create_newcase time based on the NTASKS value.   But if NTASKS is changed the PIO settings are not - this can lead to some pathologically poor IO configurations.   I propose adjusting the pio settings upon running case.setup. Since users may not always want the pio settings adjusted for them I propose an additional command line argument to case.setup --no-adjust-pio

### PR DESCRIPTION
**Pull request by jedwards4b**
Adjust the pio layout if pe layout changes

Test suite: scripts_regression_tests.py
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes #1192 

User interface changes?: case.setup has a new option --no-adjust-pio 

Code review:

### CODE REVIEW

File: scripts/Tools/case.setup [hunk 1]
```diff
@@ -57,9 +60,16 @@ def _main_func(description):
...
 
-    caseroot, clean, test_mode, reset = parse_command_line(sys.argv, description)
+    caseroot, clean, test_mode, reset, adjust_pio = parse_command_line(sys.argv, description)
     with Case(caseroot, read_only=False) as case:
-        case_setup(case, clean, test_mode, reset)
+        case_setup(case, clean=clean, test_mode=test_mode, reset=reset, adjust_pio=adjust_pio)
 
 if __name__ == "__main__":
     _main_func(__doc__)
+
```
Comment (jgfouca) [inline scripts/Tools/case.setup]: Don't need these newlines.

File: scripts/Tools/case.setup [hunk 2]
```diff
@@ -41,14 +41,17 @@ OR
 
     parser.add_argument("-t", "--test-mode", action="store_true",
                         help="Keeps the test script when the --clean argument is used")
+
     parser.add_argument("-r", "--reset", action="store_true",
                         help="Does a clean followed by setup")
 
+    parser.add_argument("--no-adjust-pio", action="store_true",
+                        help="Do NOT adjust pio settings for new pelayout")
```
Comment (jgfouca) [inline scripts/Tools/case.setup]: Maybe add just a bit of documentation to this option to explain why you might want this?

Comment (jedwards4b) [inline scripts/Tools/case.setup]: Hows this?
     help="Do NOT adjust pio settings for new pelayout."
                                  "By default if the pelayout is changed the pio layout will be adjusted."
                                  "This option overrides that feature and leaves the pio layout as currently set."

Comment (jgfouca) [inline scripts/Tools/case.setup]: That explains what the flag does, but I was looking more for an explanation of why anyone would want to do this.

Comment (jedwards4b) [inline scripts/Tools/case.setup]: I'm not sure why someone would want to do this but I'm sure they would complain if they wanted to and couldn't.

Comment (jgfouca) [inline scripts/Tools/case.setup]: OK, I guess it's fine as is then. Just take care of the whitespaces and I'll merge.

**Comment by jedwards4b** [comment]
done

**Pull request merged**